satellite/metainfo/metainfo.go

```diff
@@ -350,13 +350,28 @@ func (endpoint *Endpoint) createOrderLimitsForSegment(ctx context.Context, point
 	var combinedErrs error
 	var limits []*pb.AddressedOrderLimit
 	for _, piece := range pointer.GetRemote().GetRemotePieces() {
-		derivedPieceID := rootPieceID.Derive(piece.NodeId)
+		node, err := endpoint.cache.Get(ctx, piece.NodeId)
+		if err != nil {
+			endpoint.log.Debug("error getting node from overlay cache", zap.Error(err))
```

Comment:
maybe log nodeID? I see offline node IDs are logged below, maybe for different reasons, but thought I'd mention.

Reply:
Remember that the error returned by `cache.Get` is `overlay.ErrNodeNotFound` and it already includes the node id?

Reply:
oh shoot, you're right.